src/subspace_store.c

```diff
@@ -82,13 +85,14 @@ subspace_store_add(SubspaceStore *store, const Hypercube *hc,
 		{
 			DimensionSlice *copy;
 
-			if (i == 0 && vec->num_slices > 0)
+			if (store->max_slices_first_dimension > 0 && i == 0 && vec->num_slices >= store->max_slices_first_dimension)
 			{
 				/*
-				 * At dimension 0 only keep one slice. This is an optimization
-				 * to prevent this store from growing too large.
+				 * At dimension 0 only keep store->max_slices_first_dimension
+				 * slices. This is to prevent this store from growing too
+				 * large. Always delete the oldest.
```

Comment:
I don't fully get this "delete the oldest" comment. Is it because we remove the first entry of the dimension vector, which is sorted on ranges/slices? What if this is a hash-partitioned dimension? This could use some clarification.